lib/SPIRV/SPIRVReader.cpp

```diff
@@ -666,117 +666,106 @@ void SPIRVToLLVM::setLLVMLoopMetadata(const LoopInstType *LM,
     assert(NumParam <= LoopControlParameters.size() &&
            "Missing loop control parameter!");
   }
-  if (LC & LoopControlExtendedControlsMask) {
-    while (NumParam < LoopControlParameters.size()) {
-      switch (LoopControlParameters[NumParam]) {
-      case InitiationIntervalINTEL: {
-        // To generate a correct integer part of metadata we skip a parameter
-        // that encodes name of the metadata and take the next one
-        Metadata.push_back(llvm::MDNode::get(
-            *Context,
-            getMetadataFromNameAndParameter(
-                "llvm.loop.ii.count", LoopControlParameters[++NumParam])));
-        break;
-      }
-      case MaxConcurrencyINTEL: {
-        Metadata.push_back(llvm::MDNode::get(
-            *Context, getMetadataFromNameAndParameter(
-                          "llvm.loop.max_concurrency.count",
-                          LoopControlParameters[++NumParam])));
-        break;
+  if (LC & InitiationIntervalINTEL) {
+    // To generate a correct integer part of metadata we skip a parameter
+    // that encodes name of the metadata and take the next one
+    Metadata.push_back(llvm::MDNode::get(
+        *Context, getMetadataFromNameAndParameter(
+                      "llvm.loop.ii.count", LoopControlParameters[NumParam])));
+    ++NumParam;
+    assert(NumParam <= LoopControlParameters.size() &&
```

Comment:
I believe relation operator should be `<`, because you can only access elements in range `[0, LoopControlParameters.size())`. Can be fixed in a separate NFC patch in all places at once

Reply:
Nice catch. I'd actually prefer fixing this within this PR!

Reply:
Actually, there's more to it: the assertions should be moved to stand before the accesses to `LoopControlParameters`' elements.

Reply:
Or **actually**, I wouldn't. I'll add a `TODO` here though.